doc/timing.adoc

```diff
@@ -0,0 +1,222 @@
+= Timing
+include::../include/config.adoc[]
+include::../include/types.adoc[]
+:description: Timing "event loop" in Minetest
+:keywords: timing, event, loop, globalstep, step, after
+
+Minetest's Lua API can be used to get time and date or to schedule "events" to run once or every server step.
+
+== Settings
+
+On servers, the `dedicated_server_step` setting controls the time between server steps, which is the maximum granularity of simulation achievable without a busywait. In singleplayer, framerate determines server steps.
+
+== Lua builtins
+
+Not restricted by mod security, these functions are available to both SSMs and CSMs and allow getting times / dates.
+
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.clock[`os.clock`] - Approximate seconds CPU time used.
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.time[`os.time`] - The current time. Will usually be a UNIX timestamp in seconds.
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.difftime[`os.difftime`] - Difference between two `os.time` timestamps (usually simply `b - a`).
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.date[`os.date`] - Getting & formatting dates.
+
+== Functions
+
+=== `minetest.get_us_time`
+
+==== Usage
+
+[source,lua]
+----
+time = minetest.get_us_time()
+----
+
+==== Returns
+[%autowidth, frame=none]
+|===
+| `time` | integer | System-dependent timestamp in microseconds since an arbitrary starting point (µs)
+|===
+
+TIP: Divide by `1e6` to convert `time` into seconds.
+
+WARNING: Do not persistently store the returned `time` as it's not portable - keep it only in memory to use it while the game is running.
```

Comment:
```suggestion
CAUTION: The returned `time` is not portable across platforms - keep it only in memory for use while the game is running.
```

Reply:
Not only across platforms. It literally isn't portable at all, and not relative to a fixed point in time. Agreed on the "CAUTION" though.

Reply:
Better wording?